enclave/src/lib.rs

```diff
@@ -425,6 +425,79 @@ pub unsafe extern "C" fn accept_pdex_accounts(
     sgx_status_t::SGX_SUCCESS
 }
 
+extern "C" {
+    fn ocall_send_nonce(
+        ret_val: *mut sgx_status_t,
+        account_encoded: *const u8,
+        account_size: u32,
+        nonce: u32,
+    ) -> sgx_status_t;
+
+    pub fn ocall_send_balances(
+        ret_val: *mut sgx_status_t,
+        account_encoded: *const u8,
+        account_size: u32,
+        token_encoded: *const u8,
+        token_size: u32,
+        free: *mut u8,
+        reserved: *mut u8,
+        balance_size: u32,
+    ) -> sgx_status_t;
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn run_db_thread() -> sgx_status_t {
+    if create_in_memory_channel_storage().is_err() {
+        return sgx_status_t::SGX_ERROR_UNEXPECTED;
+    }
+
+    let mutex = if let Ok(mutex) = load_receiver() {
```

Comment:
This solution is ok, but a nicer one would be
```
... .unwrap_or_else(|e| { 
        error!("Failed to load mutex");
        return sgx_status_t::SGX_ERROR_UNEXPECTED;
})
```
(something like [here](https://github.com/capsule-corp-ternoa/sgx-keyvault/blob/develop/enclave/src/ternoa/nft_registry.rs#L104-L110))

Reply:
That wouldn't work as it would have a type conflict.